lib/characteristic.dart

```diff
@@ -20,7 +20,7 @@ class Characteristic extends InternalCharacteristic {
   /// The [Service] containing this characteristic.
   Service service;
 
-  ManagerForCharacteristic _manager;
+  final ManagerForCharacteristic _manager;
 
   /// The UUID of this characteristic.
   String uuid;
```

Comment:
Like above, also final 🙂

Reply:
Do we want any of the fields of this object (POD) to be mutable? in other words, does the user ever want to alter any of the fields of a retrieved Characteristic object, if anything it only leads to inconsistency innit?

Reply:
I wrote about it since you've finaled things on descriptor, but not on Characteristic. What needs to be done is actually move all the implementation away from characteristic to internal characteristic and make Characteristic abstract with everything final, so that it's only an interface, but it's somewhere on my to-do. Either way no worries, these are just nitpicks and I'll go over the all the classes before release, you can safely ignore it in this PR.